autorest/azure/token.go

```diff
@@ -301,14 +301,17 @@ func (spt *ServicePrincipalToken) refreshInternal(resource string) error {
 	}
 
 	var newToken Token
+	var aadError AADServiceError
 	err = autorest.Respond(resp,
 		autorest.WithErrorUnlessOK(),
-		autorest.ByUnmarshallingJSON(&newToken),
+		autorest.ByUnmarshallingSuccessandErrorJSON(&newToken, &aadError),
```

Comment:
are you sure this will work? because autorest.WithErrorUnlessOK already returns an error, so your `autorest.ByUnmarshallingSuccessandErrorJSON` most likely won't get invoked if response.Code!=200.